source/onednngraph.js

```diff
@@ -0,0 +1,515 @@
+    }
+
+    get value() {
+        return this._value;
+    }
+
+    get visible() {
+        return this._visible == false ? false : true;
+    }
+};
+
+onednngraph.Parameter = class {
+
+    constructor(name, args) {
+        this._name = name;
+        this._arguments = args;
+    }
+
+    get name() {
+        return this._name;
+    }
+
+    get visible() {
+        return true;
+    }
+
+    get arguments() {
+        return this._arguments;
+    }
+};
+
+onednngraph.Argument = class {
+
+    constructor(name, type, initializer, lt) {
+        if (typeof name !== 'string') {
+            throw new onednngraph.Error("Invalid argument identifier '" + JSON.stringify(name) + "'.");
+        }
+        this._name = name;
+        this._type = type || null;
+        this._initializer = initializer || null;
+        this._lt = lt;
+    }
+
+    get name() {
+        return this._name;
+    }
+
+    get type() {
+        if (this._initializer) {
+            return this._initializer.type;
+        }
+        return this._type;
+    }
+
+    get initializer() {
+        return this._initializer;
+    }
+
+    get description() {
+        const stride = this._lt.stride ? ('[' + this._lt.stride.map((stride) => stride ? stride.toString() : '?').join(',') + ']') : '';
+        const layout_type = this._lt.layout_type;
+        const property_type = this._lt.property_type;
+        return 'stride = ' + stride + ', layout_type = ' + layout_type + ', property_type = ' + property_type;
+    }
+};
+
+onednngraph.TensorType = class {
+
+    constructor(dataType, shape) {
+        this._dataType = dataType || '?';
+        this._shape = shape;
+    }
+
+    get dataType() {
+        return this._dataType;
+    }
+
+    get shape() {
+        return this._shape;
+    }
+
+    toString() {
+        return this._dataType + this._shape.toString();
+    }
+};
+
+onednngraph.TensorShape = class {
+
+    constructor(dimensions) {
+        this._dimensions = dimensions;
+    }
+
+    get dimensions() {
+        return this._dimensions;
+    }
+
+    toString() {
+        return this._dimensions ? ('[' + this._dimensions.map((dimension) => dimension ? dimension.toString() : '?').join(',') + ']') : '';
+    }
+};
+
+onednngraph.LogicalTensor = class {
+
+    constructor(id, dataType, shape, stride, layout_type, property_type) {
+        this._id = id;
+        this._shape = shape;
+        this._dataType = dataType;
+        this._stride = stride;
+        this._layout_type = layout_type;
+        this._property_type = property_type;
+    }
+
+    get id() {
+        return this._id;
+    }
+
+    get dataType() {
+        return this._dataType;
+    }
+
+    get shape() {
+        return this._shape;
+    }
+
+    get stride() {
+        return this._stride;
+    }
+
+    get layout_type() {
+        return this._layout_type;
+    }
+
+    get property_type() {
+        return this._property_type;
+    }
+};
+
+onednngraph.Metadata = class {
```

Comment:
Delete. This code is not used. If needed, `context.metadata(...)` will provide this standard implementation.

Reply:
changed.

Reply:
`onednn-metadata.json` contains invalid category names, check other metadata files for reference. `Convolution` is `Layer`. `AvgPool` is `Pool`. `lower_precision` and `misc` are not valid categories.

Reply:
updated and node type schema has been enabled. Now, some ops in the graph should be rendered with defined css style.

Reply:
Please also update the ordering of properties to match other `-metadata.json` files, e.g. `name` before `category`.

Reply:
done.

Reply:
Order is `name`, `category`, `inputs`, ... in other files.

Reply:
seems category is the last in onnx-metafile... Anyway, can change it. What't the impact of order?